### ISSUE TITLE
Secrets Manager Action not Working- Mode used "cloudtrail"

### ISSUE THREAD

**Issue by gulati-diksha** [cloud-custodian/cloud-custodian#6295]
**Describe the bug**
While I deploy a secrets manager policy with mode cloudtrail and give an action delete, the policy is not deleting the newly created secret. As it is not able to locate the resource based on the secret name provided.

**To Reproduce**
1. Deploy the below policy.

```
---
policies:
  - name: r-sm-tags
    resource: aws.secrets-manager
    description: >
      Event: Secrets Manager. Compliance: Missing tags. Remediation: Tag Resource
    mode:
      type: cloudtrail
      events:
        - source: secretsmanager.amazonaws.com
          event: CreateSecret
          ids: requestParameters.name
      timeout: 200
      delay: 40
      role: "arn:aws:iam::<accNumber>:role/Cloud_Custodian_Role"
    filters:
      - and:
          - or:
              - tag:CC: absent
              - tag:RS: absent
              - tag:AppDName: absent
          - tag:SMTags: absent
    actions:
      - type: tag
        tags:
          NonCompliant: true

```

2. Create a new secret.
3. Check if the action tag works on the new resource.

**Expected behavior**
If the new secret have the required tags, it should be tagged with NonCompliant: true.

**Background (please complete the following information):**
 - Python Version: 3.8.5
 - Custodian Version: 0.9.8
 - Cloud Provider: aws
 - `custodian version --debug` output
Custodian:   0.9.8
Python:      3.8.5 (default, Jul 21 2020, 10:42:08) 
             [Clang 11.0.0 (clang-1100.0.33.17)]
Platform:    posix.uname_result(sysname='Darwin', nodename='f8ffc201b550.ant.amazon.com', release='18.7.0', version='Darwin Kernel Version 18.7.0: Mon Aug 31 20:53:32 PDT 2020; root:xnu-4903.278.44~1/RELEASE_X86_64', machine='x86_64')
Using venv:  True
Docker: False
Installed: 

argcomplete==1.12.1
attrs==20.2.0
boto3==1.16.10
botocore==1.19.10
importlib-metadata==1.7.0
jmespath==0.10.0
jsonpickle==1.3
jsonschema==3.2.0
pyrsistent==0.17.3
python-dateutil==2.8.1
pyyaml==5.3.1
s3transfer==0.3.3
setuptools==47.1.0
six==1.15.0
tabulate==0.8.7
urllib3==1.25.11
zipp==3.4.0

**Comment by kapilt** [comment]
fix merged into trunk via #6296

**Comment by roiko2** [comment]
> fix merged into trunk via #6296

Im not sure this change was good, reverting your changes fixed the issue for me.
The cloudtrail event doesn't have secret name for all events... and does have arn
@kapilt  Please look [here](https://github.com/cloud-custodian/cloud-custodian/issues/9254)
